download-utils/src/lib.rs

```diff
@@ -299,3 +305,91 @@ pub fn download_snapshot<'a, 'b>(
         desired_snapshot_hash.0, rpc_addr
     ))
 }
+
+pub fn download_full_snapshot<'a, 'b>(
+    rpc_addr: &SocketAddr,
+    snapshot_config: &SnapshotConfig,
+    desired_full_snapshot_archive_info: &FullSnapshotArchiveInfo,
```

Comment:
Both `FullSnapshotArchiveInfo` and `IncrementalSnapshotArchiveInfo` implement `SnapshotArchiveInfoGetter`. Is there a way to leverage that to coalesce `download_full_snapshot` and `download_incremental_snapshot` into a single function that constructs  a `destination_path` and downloads it?

Reply:
Done. I removed this version of the download and then refactored the existing `download_snapshot()` to work for both Full and Incremental snapshots.